meta/store.go

```diff
@@ -1664,6 +1772,33 @@ func (fsm *storeFSM) Apply(l *raft.Log) interface{} {
 	return err
 }
 
+func (fsm *storeFSM) applyRemovePeerCommand(cmd *internal.Command) interface{} {
+	ext, _ := proto.GetExtension(cmd, internal.E_RemovePeerCommand_Command)
+	v := ext.(*internal.RemovePeerCommand)
+
+	id := v.GetID()
+	addr := v.GetAddr()
+
+	// Only do this if you are the leader
+	if fsm.raftState.isLeader() {
+		//Remove that node from the peer
+		fsm.Logger.Printf("removing peer for node id %d, %s", id, addr)
+		if err := fsm.raftState.removePeer(addr); err != nil {
+			fsm.Logger.Printf("error removing peer: %s", err)
+		}
+	}
+
+	// If this is the node being shutdown, stop raft
+	if fsm.id == id {
+		fsm.Logger.Printf("shutting down raft for %s", addr)
+		if err := fsm.raftState.shutdown(); err != nil {
```

Comment:
I'm leery to introduce a `non-raft` state on top of the raft states that already exist.  I do agree that something further needs to be done, but it likely won't be in this PR.  For now, the quickest thing to do is just shut it down so it doesn't try to talk to the other servers it still has an RPC connection to.